ethportal-api/src/types/consensus/beacon_state.rs

```diff
@@ -0,0 +1,312 @@
+    pub previous_version: [u8; 4],
+    #[serde(with = "serde_utils::bytes_4_hex")]
+    pub current_version: [u8; 4],
+    pub epoch: Epoch,
+}
+
+#[derive(
+    Clone, Debug, Copy, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+#[serde(transparent)]
+pub struct Epoch(#[serde(deserialize_with = "as_u64")] u64);
+
+impl Encode for Epoch {
+    fn is_ssz_fixed_len() -> bool {
+        <u64 as Encode>::is_ssz_fixed_len()
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        self.0.ssz_append(buf)
+    }
+
+    fn ssz_fixed_len() -> usize {
+        <u64 as Encode>::ssz_fixed_len()
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        0_u64.ssz_bytes_len()
+    }
+}
+
+impl Decode for Epoch {
+    fn is_ssz_fixed_len() -> bool {
+        <u64 as Decode>::is_ssz_fixed_len()
+    }
+
+    fn ssz_fixed_len() -> usize {
+        <u64 as Decode>::ssz_fixed_len()
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        Ok(Self(u64::from_ssz_bytes(bytes)?))
+    }
+}
+
+impl TreeHash for Epoch {
+    fn tree_hash_type() -> tree_hash::TreeHashType {
+        tree_hash::TreeHashType::Basic
+    }
+
+    fn tree_hash_packed_encoding(&self) -> tree_hash::PackedEncoding {
+        self.0.tree_hash_packed_encoding()
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        32usize.wrapping_div(8)
+    }
+
+    fn tree_hash_root(&self) -> Hash256 {
+        Hash256::from_slice(&int_to_fixed_bytes32(self.0))
+    }
+}
+
+/// Returns `int` as little-endian bytes with a length of 32.
+pub fn int_to_fixed_bytes32(int: u64) -> [u8; 32] {
+    let mut bytes = [0; 32];
+    let int_bytes = int.to_le_bytes();
+    bytes[0..int_bytes.len()].copy_from_slice(&int_bytes);
+    bytes
+}
+
+/// Information about a `BeaconChain` validator.
+///
+/// Spec v0.12.1
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode, TreeHash)]
+pub struct Validator {
+    pub pubkey: PubKey,
+    pub withdrawal_credentials: H256,
+    #[serde(deserialize_with = "as_u64")]
+    pub effective_balance: u64,
+    pub slashed: bool,
+    pub activation_eligibility_epoch: Epoch,
+    pub activation_epoch: Epoch,
+    pub exit_epoch: Epoch,
+    pub withdrawable_epoch: Epoch,
+}
+
+#[cfg(test)]
+#[allow(clippy::unwrap_used)]
+mod test {
+    use super::*;
+    use ::ssz::Encode;
+    use rstest::rstest;
+    use serde_json::Value;
+
+    #[rstest]
+    #[case("case_0")]
+    #[case("case_1")]
+    fn serde_beacon_state_bellatrix(#[case] case: &str) {
+        let value = std::fs::read_to_string(format!(
+            "../test_assets/beacon/bellatrix/BeaconState/ssz_random/{case}/value.yaml"
+        ))
+        .expect("cannot find test asset");
+        let value: Value = serde_yaml::from_str(&value).unwrap();
+        let content: BeaconStateBellatrix = serde_json::from_value(value.clone()).unwrap();
+        let serialized = serde_json::to_value(content).unwrap();
```

Comment:
Uhhh, I guess this works, but should we just use `serde_yaml::to_value` for consistency?

Reply:
No, because we can't compare `serde_yaml::Value` to `JsonValue` and we want to test json ser/de.